Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs

```diff
@@ -160,6 +160,50 @@ public async Task<IList<Organisation>> Get(string? type)
         return await result.AsSingleQuery().ToListAsync();
     }
 
+    public async Task<IList<Organisation>> GetPaginated(PartyRole role, int limit, int skip)
+    {
+        IQueryable<Organisation> result = context.Organisations
+            .Include(o => o.ReviewedBy)
+            .Include(o => o.Identifiers)
+            .Include(o => o.BuyerInfo)
+            .Include(o => o.SupplierInfo)
+            .Include(o => o.Addresses)
+            .ThenInclude(p => p.Address);
+
+        switch (role)
+        {
+            case PartyRole.Buyer:
+                result = result.Where(o => o.Roles.Contains(PartyRole.Buyer) || o.PendingRoles.Contains(PartyRole.Buyer));
+                break;
+            default:
+                result = result.Where(o => o.Roles.Contains(PartyRole.Tenderer));
+                break;
+        }
+
+        return await result
+            .AsSingleQuery()
+            .Skip(skip)
+            .Take(limit)
+            .ToListAsync();
+    }
+
+    public async Task<int> GetTotalCount(PartyRole role)
+    {
+        IQueryable<Organisation> result = context.Organisations;
+
+        switch (role)
+        {
+            case PartyRole.Buyer:
+                result = result.Where(o => o.Roles.Contains(PartyRole.Buyer) || o.PendingRoles.Contains(PartyRole.Buyer));
+                break;
+            default:
+                result = result.Where(o => o.Roles.Contains(PartyRole.Tenderer));
+                break;
```

Comment:
I think these API endpoints should take an array of `roles[]`, and the condition in the query would be like:

```
.Where(o => o.Roles.Any(role => roles.Contains(role)))
```

and just remove the switch entirely.

Currently this means that if you query for a list of orgs with the Buyer role, you get pending buyers back as well.